### DISCUSSION TITLE
Segmentation fault in upscayl-bin on Mac

### DISCUSSION THREAD

**Discussion by brian-t-allen** [upscayl/upscayl#695] [Q&A]
I've compiled `upscayl-ncnn` on my MacBook Pro (16-inch, 2021 with 16 GB of RAM) following the instructions. It compiles successfully and I can run it to get the instructions (`upscayl-bin -h`), but when I try to run it I get a segmentation fault.

This is using `upscayl-ncnn 2024.01.27` and `VulkanSDK 1.3.261.1`. I also tried with the most recent VulkanSDK with the same results.

I found the bundled CLI at `/Applications/Upscayl.app/Contents/Resources/bin/upscayl-bin` and I also get the segmentation fault when using it.

Upscayl.app works fantastic (thank you for creating it!), but I'd like to automate the process as part of a pipeline.

Any suggestions on resolving this?

**Comment by NayamAmarshe** [comment]
Anything not in release is experimental and can possibly fail to work. You should use the release from GitHub releases.

**Reply by brian-t-allen** [reply]
Thank you. I totally missed that you provided binaries! That's much easier than compiling.

Unfortunately it segfaults as well. I'm on Sonoma 14.3 (latest version).

I'm happy to help debug this if someone can point me in the right direction.

Thanks!

**Reply by NayamAmarshe** [reply]
Can you post the output of ./upscayl-bin?
Does it mention a -c option?

**Reply by brian-t-allen** [reply]
Here is the output. No `-c` option.

```Usage: realesrgan-ncnn-vulkan -i infile -o outfile [options]...

  -h                   show this help
  -i input-path        input image path (jpg/png/webp) or directory
  -o output-path       output image path (jpg/png/webp) or directory
  -s scale             upscale ratio (can be 2, 3, 4. default=4)
  -t tile-size         tile size (>=32/0=auto, default=0) can be 0,0,0 for multi-gpu
  -m model-path        folder path to the pre-trained models. default=models
  -n model-name        model name (default=realesr-animevideov3, can be realesr-animevideov3 | realesrgan-x4plus | realesrgan-x4plus-anime | realesrnet-x4plus)
  -g gpu-id            gpu device to use (default=auto) can be 0,1,2 for multi-gpu
  -j load:proc:save    thread count for load/proc/save (default=1:2:2) can be 1:2,2,2:2 for multi-gpu
  -x                   enable tta mode
  -f format            output image format (jpg/png/webp, default=ext/png)
  -v                   verbose output```

**Reply by NayamAmarshe** [reply]
> Here is the output. No `-c` option.
> 
> ```Usage: realesrgan-ncnn-vulkan -i infile -o outfile [options]...
> 
>   -h                   show this help
>   -i input-path        input image path (jpg/png/webp) or directory
>   -o output-path       output image path (jpg/png/webp) or directory
>   -s scale             upscale ratio (can be 2, 3, 4. default=4)
>   -t tile-size         tile size (>=32/0=auto, default=0) can be 0,0,0 for multi-gpu
>   -m model-path        folder path to the pre-trained models. default=models
>   -n model-name        model name (default=realesr-animevideov3, can be realesr-animevideov3 | realesrgan-x4plus | realesrgan-x4plus-anime | realesrnet-x4plus)
>   -g gpu-id            gpu device to use (default=auto) can be 0,1,2 for multi-gpu
>   -j load:proc:save    thread count for load/proc/save (default=1:2:2) can be 1:2,2,2:2 for multi-gpu
>   -x                   enable tta mode
>   -f format            output image format (jpg/png/webp, default=ext/png)
>   -v                   verbose output```

Strange. The segfault only happens when it has no GPU device.
I'll need to look into this

**Reply by brian-t-allen** [reply]
FYI, I tested this on the 2.9.9 release with the same results (segfault). Great release, though!

**Comment by xtremforce** [comment]
Same here. 

https://github.com/upscayl/upscayl/issues/788
